examples/exex-subscription/src/main.rs

```diff
@@ -0,0 +1,143 @@
+#![warn(unused_crate_dependencies)]
+#![allow(dead_code)]
+use alloy_primitives::{Address, U256};
+use clap::Parser;
+use futures::TryStreamExt;
+use jsonrpsee::{
+    core::SubscriptionResult, proc_macros::rpc, tracing, PendingSubscriptionSink,
+    SubscriptionMessage,
+};
+use reth_ethereum::{
+    exex::{ExExContext, ExExEvent, ExExNotification},
+    node::{api::FullNodeComponents, EthereumNode},
+};
+use std::{collections::HashMap, sync::Arc};
+use tokio::sync::{mpsc, RwLock};
+use tracing::info;
+
+/// Subscription update format
+#[derive(Debug, Clone, Copy, Default, serde::Serialize)]
+struct StorageDiff {
+    address: Address,
+    key: U256,
+    old_value: U256,
+    new_value: U256,
+}
+
+type SubscriptionMap = Arc<RwLock<HashMap<Address, Vec<mpsc::Sender<StorageDiff>>>>>;
```

Comment:
we can solve this a bit more elegantly, and replace this map with a channel, then the exex doesn't have to lock this on every block, like

```
1. subscription sends a new Subscription Message through a channel with a `oneshot::Sender` as argument
2. exex creates a new subscription and sends the receiver as response through that oneshot::Sender